fastparquet/util.py

```diff
@@ -263,6 +265,59 @@ def _get_fmd(inbytes):
     return from_buffer(data, "FileMetaData")
 
 
+def update_custom_metadata(obj, custom_metadata : dict):
+    """Update custom metadata stored in thrift object or parquet file.
```

Comment:
I have this need because during my own _update_ process for ordered datasets (as implemented in[ oups' `writer.py`](https://github.com/yohplala/oups/blob/main/oups/writer.py), and using `file_scheme='hive'`), I am tuning the `ParquetFile` object with its own methods `ParquetFile.remove_row_groups()` and `ParquetFile.write_row_groups()`.
So the `ParquetFile` instance gets modified.
To optimize processing time, for these steps, i am not writing yet the common metadata.

The scheme I have in mind is that, if new metadata needs then to be recorded, it is done only in the common metadata files, before finally writing it.

So, if this is acceptable, yes, I would like to keep it. this capitalizes in `fastparquet` library the need that to do it 'properly', you need to set `pf._kvm = None` after metadata update.
But if this is not acceptable, I will remove this part. This is not a big deal to have the line `pf._kvm = None` in `oups` indeed.